pkg/apis/networking/types.go

```diff
@@ -387,7 +387,25 @@ type IngressTLS struct {
 type IngressStatus struct {
 	// LoadBalancer contains the current status of the load-balancer.
 	// +optional
-	LoadBalancer api.LoadBalancerStatus
+	LoadBalancer IngressLoadBalancerStatus
+}
+
+// IngressLoadBalancerStatus represents the status of a load-balancer
+type IngressLoadBalancerStatus struct {
+	// Ingress is a list containing ingress points for the load-balancer.
+	// +optional
+	Ingress []IngressLoadBalancerIngress
+}
+
+// IngressLoadBalancerIngress represents the status of a load-balancer ingress point.
+type IngressLoadBalancerIngress struct {
```

Comment:
```
// LoadBalancerIngress represents the status of a load-balancer ingress point:
// traffic intended for the service should be sent to an ingress point.
type LoadBalancerIngress struct {
	// IP is set for load-balancer ingress points that are IP based
	// (typically GCE or OpenStack load-balancers)
	// +optional
	IP string

	// Hostname is set for load-balancer ingress points that are DNS based
	// (typically AWS load-balancers)
	// +optional
	Hostname string

	// Ports is a list of records of service ports
	// If used, every port defined in the service should have an entry in it
	// +optional
	Ports []PortStatus
}
```
is ok to remove the ports?

Reply:
This is a topic to discuss.  It was never intended to be under Ingress, but now it is.  I just pushed a second commit to see how it goes.

Reply:
it seems we settle on leaving the Ports and keep it the same